router/broadcast_router.go

```diff
@@ -19,22 +19,20 @@ type broadcastRouterState struct {
 	routees *actor.PIDSet
 }
 
-func (state *broadcastRouterState) SetRoutees(routees *actor.PIDSet) {
+func (state *broadcastRouterState) SetRoutees(routees actor.PIDSet) {
 	atomic.StorePointer((*unsafe.Pointer)(unsafe.Pointer(&state.routees)), unsafe.Pointer(&routees))
 }
 
-func (state *broadcastRouterState) GetRoutees() *actor.PIDSet {
-	newSet := &actor.PIDSet{}
-	routees := (*actor.PIDSet)(atomic.LoadPointer((*unsafe.Pointer)(unsafe.Pointer(&state.routees))))
-	routees.ForEach(func(i int, pid actor.PID) {
+func (state *broadcastRouterState) GetRoutees() actor.PIDSet {
```

Comment:
Why is it necessary to change the return type of `GetRoutees`? Leaving `(*randomRouterState).GetRoutees` and `(*roundRobinRouterState).GetRoutees` as they were would be fine, since the `PIDSet` isn't concurrently used for routing in those cases, and leaving the return type alone would save a few stack copies. It suffices for the `SetRoutees` methods to store atomically to `broadcastRouterState.routees`, `consistentHashRouterState.hmc`, `randomRouterState.values`, and `roundRobinRouterState.values`.